### PR TITLE
New deposit page

### LINKED ISSUE #404: [FRONTEND]  Add new page `add_deposit`
🚨 **Attention: Please read the guidelines before applying; otherwise, you will be ignored.** 🚫📋
0. Please, join the telegram group to keep in touch https://t.me/+m0IGRiBvLtsyMzIy 
1. Read guideline [here](https://gist.github.com/djeck1432/f722f5fe518c80b248ff609627da25e5)
2. Link to the Figma - [here](https://www.figma.com/design/WZGStY327tqwLAEhaIXccY/ZkLend-Position)
3. Add `add deposit` subpage in dashboard page
4. Don't add left sidebar, It will be done another contributor
5. Add mobile version of it
6. Send to backend post request `/api/add-extra-deposit/{position_id}` with `position_id` and `amount`, `token_symbol`
7. `Position_id` you can fetch in dashboard page hook
Screenshot 2024-12-18 at 18 02 47

P.S
- Please reuse [Card](https://github.com/djeck1432/spotnet/blob/main/frontend/src/components/Card/Card.jsx) component, that already in use on Dashboard page and adjust its styles according to figma. Make sure both pages still work correctly.

- Please reuse [TokenSelector](https://github.com/djeck1432/spotnet/blob/main/frontend/src/components/TokenSelector.jsx) component that already in use on Form page and adjust its styles according to figma (TokenSelector.jsx should be put in its own folder TokenSelector and tokenSelector.css should be created with isolated styles for that components). Make sure both pages still work correctly.

### PR DESCRIPTION
**Pull request by GoSTEAN**
Add new page add_deposit 
close #404

### CODE REVIEW

**Comment by whateverfw** [comment]
@GoSTEAN nothing changes in this PR, also none of the comments were resolved

**Comment by GoSTEAN** [comment]
@whateverfw Pls which comment are you referring to.

File: frontend/src/components/TokenSelector.jsx
Comment (whateverfw) [inline frontend/src/components/TokenSelector.jsx:1]: you shodn't have 2 token selector files, delete this one and work/use the one you've created

File: frontend/src/components/TokenSelector/TokenSelector.jsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import { ReactComponent as USDC } from 'assets/icons/borrow_usdc.svg';
+import { ReactComponent as STRK } from 'assets/icons/strk.svg';
+import './TokenSelector.css'
+const Tokens = [
+  { id: 'ethOption', component: <ETH />, label: 'ETH' },
+  { id: 'usdcOption', component: <USDC />, label: 'USDC' },
+  { id: 'strkOption', component: <STRK />, label: 'STRK' },
+];
+
+const TokenSelectorDeposit = ({ selectedToken, setSelectedToken }) => {
```
Comment (whateverfw) [inline frontend/src/components/TokenSelector/TokenSelector.jsx:12]: it should be called TokenSelector

File: frontend/src/pages/spotnet/dashboard/add-deposit/AddDeposit.jsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            className="amount-field-deposit"
+            aria-describedby="currency-symbol"
+            placeholder="0.00"
+          />
+          <span id="currency-symbol" className="currency-deposit">
+            STRK
+          </span>
+        </div>
+
+        <div className='dep-button'>
```
Comment (whateverfw) [inline frontend/src/pages/spotnet/dashboard/add-deposit/AddDeposit.jsx:51]: I don't see any integration with a backend, please refer to the issue.

**Review by whateverfw — CHANGES_REQUESTED** [review]
I left some initial comments again, I'm not able to test further, because I can't run an app now, it crashes for me, so please merge with main and fix merge conflcits.

@GoSTEAN

File: frontend/src/pages/spotnet/dashboard/add-deposit/AddDeposit.jsx (hunk 2 above)
Comment (GoSTEAN) [reply frontend/src/pages/spotnet/dashboard/add-deposit/AddDeposit.jsx:51]: @whateverfw   Pls kindly point to the APIs for the integration

Comment (whateverfw) [reply frontend/src/pages/spotnet/dashboard/add-deposit/AddDeposit.jsx:51]: https://github.com/djeck1432/spotnet/issues/404
```
6. Send to backend post request /api/add-extra-deposit/{position_id} with position_id and amount, token_symbol
7. Position_id you can fetch in dashboard page hook
```

**Comment by djeck1432** [comment]
@GoSTEAN Any updates?

File: frontend/src/components/TokenSelector.jsx
Comment (GoSTEAN) [reply frontend/src/components/TokenSelector.jsx:1]: I have deleted this file long ago, I dont know why is still reflecting on the repo. It is not reflecting on my local machine

File: frontend/src/components/TokenSelector/TokenSelector.jsx (hunk 1 above)
Comment (GoSTEAN) [reply frontend/src/components/TokenSelector/TokenSelector.jsx]: Changed from TokenSelectorDeposit to TokenSelector in the latest pushed

File: frontend/src/pages/spotnet/dashboard/add-deposit/AddDeposit.jsx (hunk 2 above)
Comment (GoSTEAN) [reply frontend/src/pages/spotnet/dashboard/add-deposit/AddDeposit.jsx:118]: Integration treated in the latest push. Pls checked

File: frontend/src/pages/spotnet/dashboard/add-deposit/AddDeposit.jsx [hunk 3]
```diff
@@ -1,21 +1,78 @@
...
+      
+      // Reset form after successful deposit
+      setAmount('0');
+      setSelectedToken('STRK');
+      
+      // Success notification
+      alert('Deposit successful!'); 
+
+    } catch (err) {
+      setError(err.message);
```
Comment (whateverfw) [inline frontend/src/pages/spotnet/dashboard/add-deposit/AddDeposit.jsx:69]: we don't need error state, just show error in a toast message in `onError` callback of `useMutation`

File: frontend/src/components/TokenSelector/TokenSelector.jsx [hunk 4]
```diff
@@ -1,31 +1,41 @@
...
   { id: 'usdcOption', component: <USDC />, label: 'USDC' },
-  { id: 'strkOption', component: <STRK />, label: 'STRK' },
+  { id: 'strkOption', component: <STRK />, label: 'STRK'},
 ];
 
-const TokenSelectorDeposit = ({ selectedToken, setSelectedToken }) => {
+const TokenSelector = ({ selectedToken, setSelectedToken, onSelect }) => {
+  const handleTokenChange = (token) => {
+    setSelectedToken(token.label);
+    if (onSelect) {
```
Comment (whateverfw) [inline frontend/src/components/TokenSelector/TokenSelector.jsx:17]: i don't think that `onSelect` is necessary as it is not used anywhere

File: frontend/src/components/TokenSelector/TokenSelector.jsx [hunk 5]
```diff
@@ -1,31 +1,41 @@
...
+
 const Tokens = [
   { id: 'ethOption', component: <ETH />, label: 'ETH' },
   { id: 'usdcOption', component: <USDC />, label: 'USDC' },
-  { id: 'strkOption', component: <STRK />, label: 'STRK' },
+  { id: 'strkOption', component: <STRK />, label: 'STRK'},
 ];
 
-const TokenSelectorDeposit = ({ selectedToken, setSelectedToken }) => {
+const TokenSelector = ({ selectedToken, setSelectedToken, onSelect }) => {
```
Comment (whateverfw) [inline frontend/src/components/TokenSelector/TokenSelector.jsx:14]: Please make sure this component used correctly inside `Form.jsx`

**Review by whateverfw — CHANGES_REQUESTED** [review]

File: frontend/src/pages/forms/Form.jsx [hunk 6]
```diff
@@ -123,7 +123,7 @@ const Form = () => {
           </p>
         )}
         <label className="token-select">Select Token</label>
-        <TokenSelector selectedToken={selectedToken} setSelectedToken={setSelectedToken} />
+        <TokenSelectorDeposit selectedToken={selectedToken} setSelectedToken={setSelectedToken} />
```
Comment (whateverfw) [inline frontend/src/pages/forms/Form.jsx:126]: TokenSelector should be used here as you renamed yoiur file.

**Review by whateverfw — CHANGES_REQUESTED** [review]
@GoSTEAN please resolve merge conflicts.

File: frontend/src/pages/spotnet/dashboard/add_deposit.jsx
Comment (whateverfw) [inline frontend/src/pages/spotnet/dashboard/add_deposit.jsx:1]: remove this file, seems like you already have another one?

File: frontend/src/App.jsx [hunk 7]
```diff
@@ -104,6 +106,7 @@ function App() {
             element={walletId ? <Navigate to="/" /> : <Login onConnectWallet={handleConnectWallet} />}
           />
           <Route path="/dashboard" element={<Dashboard telegramId={window?.Telegram?.WebApp?.initData?.user?.id} />} />
+          <Route path="/addDeposit" element={<AddDeposit />} />
```
Comment (whateverfw) [inline frontend/src/App.jsx:109]: it should be /add-deposit

File: frontend/src/components/TokenSelector/TokenSelector.jsx
Comment (whateverfw) [inline frontend/src/components/TokenSelector/TokenSelector.jsx:1]: styles do not match figma.

**Review by whateverfw — CHANGES_REQUESTED** [review]

**Comment by djeck1432** [comment]
Any updates? @GoSTEAN

**Comment by GoSTEAN** [comment]
Working on it

File: frontend/src/pages/spotnet/dashboard/add-deposit/AddDeposit.jsx [hunk 8]
```diff
@@ -1,21 +1,78 @@
...
     }
   };
 
+  const handleCancel = () => {
+    setAmount('0');
+    setSelectedToken('STRK');
+    setError(null);
+  };
+
+  const handleDeposit = async () => {
```
[the remark this answers is not in the document]
Comment (GoSTEAN) [reply frontend/src/pages/spotnet/dashboard/add-deposit/AddDeposit.jsx:31]: @whateverfw There is no custom hook in the hook folder , I will be creating it that is customHook.js

**Comment by djeck1432** [comment]
@GoSTEAN you should have done yesterday.
waiting and will do last code review, if not everything is ready you will be merged and get partially reward only

**Comment by GoSTEAN** [comment]
Pushing in the next 15minutes

**Comment by GoSTEAN** [comment]
@djeck1432 Pls i dont have any conflicting file

File: frontend/src/components/TokenSelector.jsx
Comment (whateverfw) [reply frontend/src/components/TokenSelector.jsx:1]: ok nevermind then, will delete it later.

File: frontend/src/pages/spotnet/dashboard/add-deposit/AddDeposit.jsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (whateverfw) [reply frontend/src/pages/spotnet/dashboard/add-deposit/AddDeposit.jsx]: yes, of course you should create it and call `useAddDeposit`

**Review by whateverfw — CHANGES_REQUESTED** [review]
Pleas resolve comments and fix merge conflicts. Pull latest changes from main branch, then try to merge it to your branch. I just tried myself  it and see merge conflicts locally.

File: frontend/src/hooks/customHook.js
[the remark this answers is not in the document]
Comment (GoSTEAN) [reply frontend/src/hooks/customHook.js:1]: Screenshot from 2024-12-22 12-47-48
 I have synced with the main and git pull. This is my terminal

**Comment by djeck1432** [comment]
@GoSTEAN Tried at least reward you partially, sent you butch of messages, any response, I should close this pr,
Feel free to apply next time

**Pull request closed without merging**